obspy/mseed/src/libmseed/msrutils.c

```diff
@@ -358,7 +358,16 @@ msr_normalize_header ( MSRecord *msr, flag verbose )
 	  sec = msr->starttime / (HPTMODULUS / 10000);
 	  usec = msr->starttime - (sec * (HPTMODULUS / 10000));
 	  usec /= (HPTMODULUS / 1000000);
-	  
+
+      /* Deal with negative timestamps and a negative offset.
```

Comment:
It looked right in the editor I used. I couldn't use my normal editor as it auto-formats a bunch of things so the changelog would have been huge...